molvis/molvisbase/include/inviwo/molvisbase/util/molvisutils.h

```diff
@@ -0,0 +1,256 @@
+    0xFF007D00, 0xFFFAAB70, 0xFFFFBA00, 0xFFFFA100, 0xFFFF8F00, 0xFFFF8000, 0xFFFF6B00, 0xFFF25C54,
+    0xFFE35C78, 0xFFE34F8A, 0xFFD436A1, 0xFFD41FB3, 0xFFBA1FB3, 0xFFA60DB3, 0xFF870DBD, 0xFF6600C7,
+    0xFF5900CC, 0xFF4F00D1, 0xFF4500D9, 0xFF3800E0, 0xFF2E00E6, 0xFF2600EB, 0xFF2200F0, 0xFF2000F6,
+    0xFF1E00F8, 0xFF1C00FA, 0xFF1A00FC, 0xFF1800FD, 0xFF1600FE, 0xFF1400FF, 0xFF1200FF,
+};
+
+}  // namespace detail
+
+// Element functions
+constexpr Element element(int atomicNumber) { return static_cast<Element>(atomicNumber); }
+constexpr const char* name(Element symbol) { return detail::names[(int)symbol]; }
+constexpr const char* symbol(Element symbol) { return detail::symbols[(int)symbol]; }
+constexpr vec4 color(Element symbol) {
+    unsigned int c = detail::colorsRasmolCPKnew[static_cast<int>(symbol)];
+    return vec4{(c >> 0) & 0xff, (c >> 8) & 0xff, (c >> 16) & 0xff, (c >> 24) & 0xff} / 255.0f;
+}
+constexpr double vdwRadius(Element symbol) { return detail::vdw_radii[(int)symbol]; }
+constexpr double covalentRadius(Element symbol) { return detail::covalent_radii[(int)symbol]; }
+constexpr double atomicMass(Element symbol) { return detail::atomic_mass[(int)symbol]; }
+
+constexpr Element fromFullName(std::string_view fullAtomName) {
+    if (fullAtomName.empty()) return Element::Unknown;
+
+    // Prune
+    auto it = fullAtomName.begin();
+    while ((it != fullAtomName.end()) && !std::isalpha(*it)) {
+        ++it;
+    }
+    fullAtomName.remove_prefix(std::distance(fullAtomName.begin(), it));
+
+    if (fullAtomName.empty()) {
+        return Element::Unknown;
+    }
+
+    auto it2 = fullAtomName.rbegin();
+    while ((it2 != fullAtomName.rend()) && !std::isalpha(*it2)) {
+        ++it2;
+    }
+    fullAtomName.remove_suffix(std::distance(fullAtomName.rbegin(), it2));
+
+    // Two or more (Try to match first two)
+    if (fullAtomName.size() > 1) {
+        for (int i = 1; i < num_elements; i++) {
+            if (fullAtomName.compare(0, 2, symbol(static_cast<Element>(i))) == 0) {
+                return static_cast<Element>(i);
+            }
+        }
+    }
+    // Try to match only against first character
+    for (int i = 1; i < num_elements; i++) {
+        std::string_view elem = symbol(static_cast<Element>(i));
+        if (elem.size() == 1 && fullAtomName[0] == elem[0]) {
+            return static_cast<Element>(i);
+        }
+    }
+    // Try to match against two characters with the latter in lower case
+    if (fullAtomName.size() > 1) {
+        for (int i = 1; i < num_elements; i++) {
+            std::string_view elem = symbol(static_cast<Element>(i));
+            if (elem.size() == 2 && fullAtomName[0] == elem[0] &&
+                std::tolower(fullAtomName[1]) == elem[1]) {
+                return static_cast<Element>(i);
+            }
+        }
+    }
+
+    return Element::Unknown;
+}
+
+}  // namespace element
+
+bool IVW_MODULE_MOLVISBASE_API covalentBondHeuristics(Element element1, const dvec3& pos1,
```

Comment:
What kind of heuristics?

Reply:
Based on distance. It's on of the standard methods for determining covalent bonds.

Reply:
added more information